src/freenet/support/io/SkipShieldingInputStream.java

```diff
@@ -46,6 +46,14 @@ public SkipShieldingInputStream(InputStream in) {
 
     @Override
     public long skip(long n) throws IOException {
-        return n < 0 ? 0 : read(SKIP_BUFFER, 0, (int) Math.min(n, SKIP_BUFFER_SIZE));
+        int retval;
```

Comment:
Please don't add a variable just for the return value; returning the final value from where it's found is preferred.

Reply:
Well the whole point of this patch is to check the return value of that read() call to make sure it's not negative. How can that be done without introducing a return variable?

Reply:
```
if (n < 0) {
  return 0;
}

Reply:
Of course, if you _need_ to store a return value, go right ahead but just for the sake of “only have a single exit point” (which is quite stupid)? Please don’t.